### PR TITLE
feat: Add benchmark example

### PR DESCRIPTION
**Pull request by jidicula**
Adds a simple benchmark example and its invocation.

Related to #162

### CODE REVIEW

File: examples/testing/main_test.sh [hunk 1]
```diff
@@ -16,3 +16,29 @@ $ go test -v
     --- PASS: TestIntMinTableDriven/-1,0 (0.00s)
 PASS
 ok  	examples/testing	0.023s
+
+# Run all benchmarks in the current project in verbose
+# mode. All tests are run prior to benchmarks. The `bench`
+# flag receives a regex for benchmark function names.
+$ go test -v -bench=.
+=== RUN   TestIntMinBasic
```
Comment (eliben) [inline examples/testing/main_test.sh:24]: I would leave the `-v` out of the benchmark example - it just repeats lots of long text

File: examples/testing/main_test.go [hunk 2]
```diff
@@ -66,3 +66,12 @@ func TestIntMinTableDriven(t *testing.T) {
 		})
 	}
 }
+
+// Benchmark tests typically go in `_test.go` files and begin with `Benchmark`.
+// Each benchmark function is run several times, with `b.N` increasing on each
+// run until the `testing` runner deems the benchmark to be accurate.
+func BenchmarkIntMin(b *testing.B) {
```
Comment (eliben) [inline examples/testing/main_test.go:73]: Hmm, I would hope for a clearer explanation of `b.N`. What the reader needs to know is that they have to run their "thing" in a loop `b.N` times

**Review by eliben — CHANGES_REQUESTED** [review]
Thanks, I think this is a useful contribution.

Unsure about the phrasing in the comment - open for discussion

File: examples/testing/main_test.sh (hunk 1 above)
Comment (jidicula) [reply examples/testing/main_test.sh]: Resolved in bf78b89

File: examples/testing/main_test.go (hunk 2 above)
Comment (jidicula) [reply examples/testing/main_test.go:74]: Hopefully addressed in bf78b89.

File: examples/testing/main_test.go [hunk 3]
```diff
@@ -66,3 +66,15 @@ func TestIntMinTableDriven(t *testing.T) {
...
 	}
 }
+
+// Benchmark tests typically go in `_test.go` files and are
+// named beginning with `Benchmark`. The `testing` runner
+// executes each benchmark function several times, increasing
+// `b.N` on each run until it collects a precise measurement.
+func BenchmarkIntMin(b *testing.B) {
+	// To measure a benchmark, the `testing` runner calls the
+	// function in a loop `b.N` times.
```
Comment (eliben) [inline examples/testing/main_test.go:76]: Hmm, this doesn't sound right either, sorry. The runner doesn't call our function in a loop - it's our code that does it.

I would say something like: "Typically the benchmark runs a function we're benchmarking in a loop `b.N` times"

**Review by eliben — CHANGES_REQUESTED** [review]

**Comment by eliben** [comment]
Please resolve conflicts

**Review by eliben — APPROVED** [review]

**Pull request merged**